dockerfiles/haskell/8.8.4/Dockerfile

```diff
@@ -0,0 +1,44 @@
+# First Stage: Builder
+FROM debian:buster-slim AS builder
+
+# Install necessary dependencies for building GHC
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+    curl build-essential libffi-dev libgmp-dev libtinfo5 ca-certificates xz-utils \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install GHCup (Haskell toolchain manager) without automatically installing the latest GHC
+RUN curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | bash -s -- -y --no-ghc \
+    && /root/.ghcup/bin/ghcup install ghc 8.10.7 \
+    && /root/.ghcup/bin/ghcup set ghc 8.10.7 \
+    && rm -rf /root/.ghcup/ghc/9.4.8
+
+# Clean up unnecessary build artifacts and cache
+RUN rm -rf /root/.ghcup/cache /var/lib/apt/lists/*
+
+
+# Second Stage: Final Image
+FROM debian:buster-slim
```

Comment:
Please use ubuntu rather than debian

Reply:
I used this specifically because it's smaller, and I think it reduces it by almost half a GB iirc

Reply:
The size difference between the two base images is very slim. I've even seen the ubuntu smaller in some cases. We prefer to use a consistent base image when possible. A difference of half a GB would likely be coming from something done inside the dockerfile.

Reply:
if you say so, but I'll have to rework some of the file because `libffi6` isn't supported in ubuntu 20+

Reply:
done